packages/ts/react-signals/test/FullStackSignal.spec.tsx

```diff
@@ -4,13 +4,50 @@ import { render } from '@testing-library/react';
 import { ConnectClient, type Subscription } from '@vaadin/hilla-frontend';
 import sinon from 'sinon';
 import sinonChai from 'sinon-chai';
-import { type StateEvent, StateEventType } from '../src/FullStackSignal.js';
-import { NumberSignal } from '../src/index.js';
+import { DependencyTrackingSignal, type StateEvent, StateEventType } from '../src/FullStackSignal.js';
+import { computed, NumberSignal } from '../src/index.js';
 import { nextFrame } from './utils.js';
 
 use(sinonChai);
 
 describe('@vaadin/hilla-react-signals', () => {
+  describe('DependencyTrackingSignal', () => {
+    class TestSignal<T = unknown> extends DependencyTrackingSignal<T> {
+      constructor(value: T | undefined, onSubscribe: () => void, onUnsubscribe: () => void) {
+        super(value, onSubscribe, onUnsubscribe);
+        this.subscribe(() => {}); // Ignores the internal subscription.
+      }
+    }
+
+    let onSubscribe: () => void;
+    let onUnsubscribe: () => void;
```

Comment:
Better to have a specific Sinon type here:
```suggestion
    let onSubscribe: sinon.SinonStub;
    let onUnsubscribe: sinon.SinonStub;
```